installer/src/main/kotlin/com/teamscale/profiler/installer/utils/TeamscaleUtils.kt

```diff
@@ -0,0 +1,116 @@
+			.build()
+
+		try {
+			client.newCall(request).execute().use { response ->
+				handleErrors(response, this)
+			}
+		} catch (e: SSLException) {
+			throw FatalInstallerError(
+				"""Failed to connect via HTTPS to ${this.url}
+				Please ensure that your Teamscale instance is reachable under ${this.url} and that it is configured for HTTPS, not HTTP. E.g. open that URL in your browser and verify that you can connect successfully.
+
+				If you want to accept self-signed or broken certificates without an error you can use --insecure.
+				""".trimIndent(), e
+			)
+		} catch (e: UnknownHostException) {
+			throw FatalInstallerError(
+				"The host ${this.url} could not be resolved. Please ensure you have no typo and that this host is reachable from this server.", e
+			)
+		} catch (e: ConnectException) {
+			throw FatalInstallerError(
+				"The host ${this.url} refused a connection. Please ensure that you have no typo and that this endpoint is reachable and not blocked by firewalls.", e
+			)
+		} catch (e: SocketTimeoutException) {
+			throw FatalInstallerError(
+				"Request timeout reached. Please ensure that you have no typo and that this endpoint is reachable and not blocked by firewalls.", e
+			)
+		} catch (e: IOException) {
+			throw FatalInstallerError(
+				"Teamscale is not reachable from this machine. Please check firewall settings.", e
+			)
+		}
+	}
+
+	@Throws(FatalInstallerError::class)
+	private fun handleErrors(response: Response, credentials: TeamscaleCredentials) {
+		if (response.isRedirect) {
+			var location = response.header("Location")
+			if (location == null) {
+				location = "<server did not provide a location header>"
+			}
+			throw FatalInstallerError(
+				("You provided an incorrect URL."
+						+ " The server responded with a redirect to " + "'" + location + "'."
+						+ " This may e.g. happen if you used HTTP instead of HTTPS."
+						+ " Please use the correct URL for Teamscale instead.")
+			)
+		}
+
+		if (response.code == 401) {
+			val editUserUrl = getEditUserUrl(credentials.url!!, credentials.username)
+			throw FatalInstallerError(
+				"""
+				You provided incorrect credentials. Either the user '${credentials.username}' does not exist in Teamscale or the access key you provided is incorrect. Please check both the username and access key in Teamscale under Admin > Users: $editUserUrl
+				Please use the user's access key, not their password.
+				""".trimIndent()
+			)
+		}
+
+		if (!response.isSuccessful) {
+			throw FatalInstallerError(
+				"Unexpected response from Teamscale, HTTP status ${response.code} ${response.message}"
+			)
+		}
+	}
+
+
+	/** Returns a URL to the edit page of the given user.  */
+	private fun getEditUserUrl(teamscaleBaseUrl: HttpUrl, username: String?) =
+		fixFragment(
+			teamscaleBaseUrl.newBuilder().addPathSegment("admin.html#users")
+				.addQueryParameter("action", "edit").addQueryParameter("username", username)
```

Comment:
While on it, would you mind updating this to the new URL `admin/users`. `fixFragment` should then no longer be necessary.